src/Microsoft.AspNet.Mvc.Core/Formatters/AcceptHeaderParser.cs

```diff
@@ -0,0 +1,179 @@
+                    else
+                    {
+                        throw new FormatException(string.Format(CultureInfo.InvariantCulture, "Invalid values '{0}'.",
+                            value.Substring(index)));
+                    }
+                }
+            }
+        }
+
+        private static bool TryParseValue(string value, ref int index, out MediaTypeSegmentWithQuality parsedValue)
+        {
+            parsedValue = default(MediaTypeSegmentWithQuality);
+
+            // If multiple values are supported (i.e. list of values), then accept an empty string: The header may
+            // be added multiple times to the request/response message. E.g.
+            //  Accept: text/xml; q=1
+            //  Accept:
+            //  Accept: text/plain; q=0.2
+            if (string.IsNullOrEmpty(value) || (index == value.Length))
+            {
+                return true;
+            }
+
+            var separatorFound = false;
+            var current = GetNextNonEmptyOrWhitespaceIndex(value, index, out separatorFound);
+
+            if (separatorFound && !true)
+            {
+                return false; // leading separators not allowed if we don't support multiple values.
+            }
+
+            if (current == value.Length)
+            {
+                if (true)
+                {
+                    index = current;
+                }
+                return true;
+            }
+
+            MediaTypeSegmentWithQuality result;
+            var length = GetMediaTypeWithQualityLength(value, current, out result);
+
+            if (length == 0)
+            {
+                return false;
+            }
+
+            current = current + length;
+            current = GetNextNonEmptyOrWhitespaceIndex(value, current, out separatorFound);
+
+            // If we support multiple values and we've not reached the end of the string, then we must have a separator.
+            if ((separatorFound && !true) || (!separatorFound && (current < value.Length)))
+            {
+                return false;
+            }
+
+            index = current;
+            parsedValue = result;
+            return true;
+        }
+
+        private static int GetNextNonEmptyOrWhitespaceIndex(
+            string input,
+            int startIndex,
+            out bool separatorFound)
+        {
+            Contract.Requires(input != null);
+            Contract.Requires(startIndex <= input.Length); // it's OK if index == value.Length.
+
+            separatorFound = false;
+            var current = startIndex + HttpTokenParsingRules.GetWhitespaceLength(input, startIndex);
+
+            if ((current == input.Length) || (input[current] != ','))
+            {
+                return current;
+            }
+
+            // If we have a separator, skip the separator and all following whitespaces. If we support
+            // empty values, continue until the current character is neither a separator nor a whitespace.
```

Comment:
This is verbatim from the parsing code in HttpAbstractions, I'm not changing any of this